### PR TITLE
Update China holidays: add observed and substituted holidays in 2001-2024

### LINKED ISSUE #1601: Update 2024 Chinese holiday
I am Chinese, and Chinese holidy is various from other years,and the holidays of 2024 is published by Chinese government,ref:https://www.gov.cn/yaowen/liebiao/202310/content_6911540.htm。
I want to update this information as my first issue good issue,Can I do this?

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Resolves #1601 

Update China holidays: add observed and substituted holidays in 2001-2024.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/locale/th/LC_MESSAGES/CN.po [hunk 1]
```diff
@@ -59,3 +68,20 @@ msgstr "วันเด็กสากล"
...
 msgstr "วันสถาปนากองทัพปลดปล่อยประชาชนจีน"
+
+#. Date format (see strftime() Format Codes).
+msgid "%Y-%m-%d"
+msgstr "%d/%m/%Y"
+
+#. Day off (substituted from %s).
+#, c-format
+msgid "休息日（%s日起取代）"
+msgstr "วันพัก (แทนที่จาก %s)"
```
Comment (PPsyrius) [inline holidays/locale/th/LC_MESSAGES/CN.po:79]: Consider changing this to "วันหยุด (แทน %s)" instead

File: holidays/locale/th/LC_MESSAGES/CN.po [hunk 2]
```diff
@@ -59,3 +68,20 @@ msgstr "วันเด็กสากล"
...
+
+#. 70th Anniversary of the Victory of the Chinese People’s War of Resistance
+#. against Japanese
+#. Aggression and the World Anti-Fascist War.
+msgid "中国人民抗日战争暨世界反法西斯战争胜利70周年纪念日"
+msgstr ""
+"ครบรอบ 70 "
```
Comment (PPsyrius) [inline holidays/locale/th/LC_MESSAGES/CN.po:86]: [The official Thai translation](https://oldweb.ditp.go.th/contents_attach/132578/132578.pdf) uses "ครบรอบ 70 ปีแห่งการได้รับชัยชนะจากสงครามต่อต้านญี่ปุ่นและลัทธิฟาสซิสต์โลก" here so we might want to go with that instead.

(they got some typo in the referenced translation above for the Thai translation of "Fascism" - corrected to [Royal Society of Thailand](https://dictionary.orst.go.th/)'s official spelling)

**Review by PPsyrius — CHANGES_REQUESTED** [review]
Thanks for improving the Chinese calendar, here are some of the Thai language translations that likely need to be changed here. 👀

**Comment by KJhellico** [comment]
@PPsyrius, thank you! It's Google Translate 🤷🏻‍♂️

File: holidays/countries/china.py [hunk 3]
```diff
@@ -326,51 +201,234 @@ class CHN(China):
...
-    mid_autumn_festival_2010_special = tr("中秋节")
+    mid_autumn_festival = tr("中秋节")
+
+    # 70th Anniversary of the Victory of the Chinese People’s War of Resistance against
+    # Japanese Aggression and the World Anti-Fascist War.
+    victory_70_anniversary = tr("中国人民抗日战争暨世界反法西斯战争胜利70周年纪念日")
 
     special_public_holidays = {
+        2001: (
+            (JAN, 29, JAN, 20),  # spring_festival
```
Comment (arkid15r) [inline holidays/countries/china.py:221]: The comments look a bit unusual, could you make them look like "Labor day" or "Spring festival" for example?

**Review by arkid15r — COMMENTED** [review]
LGTM

**Review by arkid15r — COMMENTED** [review]
BTW, does this PR close https://github.com/vacanza/python-holidays/issues/1601 ?

**Review by arkid15r — APPROVED** [review]

**Pull request merged**